File: plugins/api/src/main/java/no/lillehaug/landingsopplysninger/api/LandingsopplysningerRepository.java
package no.lillehaug.landingsopplysninger.api;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.time.LocalDate;

public interface LandingsopplysningerRepository {
    Flux<LeveringslinjeWithId> alleLeveranselinjer();
    Flux<LeveringslinjeWithId> alleLeveranselinjer(LandingsdataQuery query);
    Flux<LeveringslinjeWithId> lagreLeveranselinjer(Flux<Leveringslinje> leveringslinjer);

    Mono<LocalDate> forrigeLandingFor(String registration);
}

File: plugins/api/src/main/java/no/lillehaug/landingsopplysninger/api/LandingsopplysningerRepository.kt
package no.lillehaug.landingsopplysninger.api

import java.time.LocalDate

interface LandingsopplysningerRepository {
    fun alleLeveranselinjer(): List<LeveringslinjeWithId>
    fun alleLeveranselinjer(landingsdataQuery: LandingsdataFraTilQuery): List<LeveringslinjeWithId>
    fun alleLeveranselinjerByDates(num: Int, start: Int): List<LeveringslinjeWithId>

    fun lagreLeveranselinjer(leveranselinjer: Collection<Leveringslinje>)

    fun forrigeLandingFor(registration: String): LocalDate

}
